src/otaclient/client_package.py

```diff
@@ -286,6 +286,7 @@ def download_client_package(
             logger.exception(
                 f"failure during downloading and verifying OTA client package: {e!r}"
             )
+            raise
```

Comment:
When the manifest file is missing from S3, the exception is caught by the upper layer.
Therefore, this change does not affect the actual behavior, but it ensures that exceptions are properly raised at this point.

Reply:
Just curious, what will happen if we don't raise here 🤔

Reply:
Later step(client copy stage) try to handle the manifest/squashfs, then raises exceptions because it doesn't exist.